pkg/flow/adapter/transformation/handler.go

```diff
@@ -43,16 +68,29 @@ type ceContext struct {
 	Extensions                  map[string]interface{} `json:"Extensions,omitempty"`
 }
 
-// NewHandler creates Handler instance.
-func NewHandler(context, data []v1alpha1.Transform) (Handler, error) {
-	contextPipeline, err := newPipeline(context)
+// NewHandler satisfies pkgadapter.AdapterConstructor by creating a Handler instance.
+func NewHandler(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, _ cloudevents.Client) pkgadapter.Adapter {
```

Comment:
@tzununbekov I _think_ we should be able to use the client passed to this function as-is when `K_SINK` is set?
It's not the purpose of this PR so I decided not to touch the existing logic, but it would be interesting to know your opinion about it.

Reply:
thanks for the note, I'll take a closer look at this tomorrow